execution_engine/src/storage/trie_store/db.rs

```diff
@@ -142,147 +120,103 @@ impl Store<Digest, Trie<Key, StoredValue>> for ScratchTrieStore {
+        matches!(self, RootMigration::NotStarted)
+    }
+    /// Has the migration been partially completed
+    pub fn is_partial(&self) -> bool {
+        matches!(self, RootMigration::Partial)
+    }
+    /// Has the migration been completed
+    pub fn is_complete(&self) -> bool {
+        matches!(self, RootMigration::Complete)
+    }
+}
 
-impl RocksDbStore {
-    /// Create a new environment for RocksDB.
-    pub fn new(path: impl AsRef<Path>) -> Result<RocksDbStore, rocksdb::Error> {
-        Self::new_with_opts(path, Self::rocksdb_defaults())
+impl RocksDbTrieStore {
+    /// Create a new trie store backed by RocksDB.
+    pub fn new(path: impl AsRef<Path>) -> Result<Self, rocksdb::Error> {
+        let store = DbStore::new(path)?;
+        Ok(Self { store })
     }
 
-    /// Create a new environment for RocksDB with the specified options.
+    /// Create a new trie store for RocksDB with the specified options.
     pub fn new_with_opts(
         path: impl AsRef<Path>,
         rocksdb_opts: Options,
-    ) -> Result<RocksDbStore, rocksdb::Error> {
-        let db = Arc::new(DBWithThreadMode::<MultiThreaded>::open_cf(
-            &rocksdb_opts,
-            path.as_ref(),
-            ACTIVE_COLUMN_FAMILIES.to_vec(),
-        )?);
-
-        Ok(RocksDbStore {
-            db,
-            path: path.as_ref().to_path_buf(),
-        })
+    ) -> Result<Self, rocksdb::Error> {
+        let store = DbStore::new_with_opts(path, rocksdb_opts)?;
+        Ok(Self { store })
     }
 
-    /// Default constructor for rocksdb options.
-    pub fn rocksdb_defaults() -> Options {
-        let mut factory_opts = BlockBasedOptions::default();
-        factory_opts.set_block_size(ROCKS_DB_BLOCK_SIZE_BYTES);
-
-        let mut db_opts = Options::default();
-        db_opts.set_block_based_table_factory(&factory_opts);
-
-        db_opts.set_compression_type(ROCKS_DB_COMPRESSION_TYPE);
-        db_opts.set_compression_options(
-            ROCKS_DB_WINDOW_BITS,
-            ROCKS_DB_ZSTD_COMPRESSION_LEVEL,
-            ROCKS_DB_ZSTD_STRATEGY,
-            ROCKS_DB_ZSTD_MAX_DICT_BYTES,
-        );
-
-        // seems to lead to a sporadic segfault within rocksdb compaction
-        // const ROCKS_DB_ZSTD_MAX_TRAIN_BYTES: i32 = 1024 * 1024; // 1 MB
-        // db_opts.set_zstd_max_train_bytes(ROCKS_DB_ZSTD_MAX_TRAIN_BYTES);
-
-        db_opts.set_compaction_style(ROCKS_DB_COMPACTION_STYLE);
-        db_opts.set_max_bytes_for_level_base(ROCKS_DB_MAX_LEVEL_FILE_SIZE_BYTES);
-        db_opts.set_max_open_files(ROCKS_DB_MAX_OPEN_FILES);
-
-        db_opts.create_missing_column_families(true);
-        db_opts.create_if_missing(true);
-
-        // recommended to match # of cores on host.
-        db_opts.increase_parallelism(num_cpus::get() as i32);
-
-        db_opts.create_missing_column_families(true);
-        db_opts.create_if_missing(true);
-
-        db_opts
+    /// Access the underlying DbStore.
+    pub fn get_db_store(&self) -> &DbStore {
```

Comment:
Should we be exposing the underlying RocksDB for consumption in the public API? Can we hide it behind domain-specific methods?

Reply:
We can make this `pub(crate)` for now. Done in [6c20f4a](https://github.com/casper-network/casper-node/pull/2999/commits/6c20f4a0f076a2a375dc7d0f332871c085825545)